client/src/App.tsx

```diff
@@ -20,7 +20,7 @@ function App(): JSX.Element {
               <Switch>
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
+                <Route exact path="/dashboard" component={Dashboard}>
                   <Dashboard />
                 </Route>
```

Comment:
You're passing the Dashboard component here twice, if you use the component property you do not need to pass it as a child as well

Reply:
fixed it